controllers/telemetry/telemetrymanager_controller.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	telemetryv1alpha1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+)
+
+const (
+	requeueInterval = time.Second * 10
+	finalizer       = "telemetrymanager.kyma-project.io/finalizer"
+	fieldOwner      = "telemetrymanager.kyma-project.io/owner"
+)
+
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	*rest.Config
+	// EventRecorder for creating k8s events
+	record.EventRecorder
+}
+
+func NewTelemetryManagerReconciler(client client.Client, scheme *runtime.Scheme, eventRecorder record.EventRecorder) *Reconciler {
+	var r Reconciler
+	r.Client = client
+	r.Scheme = scheme
+	r.EventRecorder = eventRecorder
+	return &r
+}
+
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=telemetrymanagers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=telemetrymanagers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=telemetrymanagers/finalizers,verbs=update
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	objectInstance := telemetryv1alpha1.TelemetryManager{}
```

Comment:
I think we should move the reconciliation logic into the `internal/reconciler` dir to be consistent with the changes done here #64